### PR TITLE
Refactor Notification-controller API code

### PR DESCRIPTION
**Pull request by nirarg**
This Change is the first step of refactoring and fixing notification-service API
This change include the following:

- Change notification-controller:
Change Get function to return "ResponseEntity<Page<NotificationRecordResponseDTO>>" instead of
    "PagedModel<NotificationRecordResponseDTO>" in order to fix miss matching between the returned value
    to the structure defined in swagger.
Fixes: [FLPATH-260](https://issues.redhat.com/browse/FLPATH-260)
- Change controller unit tests following the changes
- Introduce new notification-service integration test
Currently this include one test validate basic functionality
(related to: [FLPATH-273](https://issues.redhat.com/browse/FLPATH-273))

### CODE REVIEW

File: .github/workflows/test.yml
Comment (pkliczewski) [inline .github/workflows/test.yml:39]: Why do we want to set `failIfNoTests` to false?

File: Makefile
Comment (pkliczewski) [inline Makefile:1]: why do we want to set `failIfNoTests` to false in this file?

File: notification-service-sdk/.openapi-generator/VERSION
Comment (pkliczewski) [inline notification-service-sdk/.openapi-generator/VERSION:1]: this change makes it inconsistent with workflow-service-sdk. Please update both

File: notification-service-sdk/pom.xml
Comment (pkliczewski) [inline notification-service-sdk/pom.xml:197]: please make it consistent with workflow-service-sdk

File: notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java
Comment (eloycoto) [inline notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java:90]: Should we have a method on `NotificationRecordResponseDTO.fromNotificationRecord(NotificationRecord)` we moved to that way in the workflow-service.

File: notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java
Comment (eloycoto) [inline notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java:99]: Return the number

File: notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java
Comment (eloycoto) [inline notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java:118]: TBH, I prefer to use the toModel() is much better for testing.

File: workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java
Comment (eloycoto) [inline workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java:55]: we shouldn't use try in test, should be assertDoesNotThrow

File: workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java
Comment (eloycoto) [inline workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java:68]: all the try here shouldn't be here.

File: workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java
Comment (eloycoto) [inline workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java:132]: Is this really needed? if we need to log in a test maybe can be splitted in two?

**Review by eloycoto — CHANGES_REQUESTED** [review]

File: notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java
Comment (nirarg) [reply notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java:90]: I saw that was the way used both for `workflowDefinition` and `project` list (using modelMapper)
If we want to change it, lets address this change for all modules in another PR
(Stop using modelMapper)

File: workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java
Comment (nirarg) [reply workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java:55]: `assertDoesNotThrow` is supported only since junit 5.2, while we use junit 4.13.1
As far as I understand, this can't be updated yet because of Maven supprt
Am I wrong?

Comment (eloycoto) [reply workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java:55]: we used assertDoesNotThrow in the workflow-service, are u sure about?

File: workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java
Comment (eloycoto) [reply workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java:132]: at the moment you need to add logs inside the test, means that the log may have too many complexity and maybe you can split in more.

File: notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java
Comment (eloycoto) [reply notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java:90]: Why need to be addressed in a different issue?

What I mentioned is to use something like this, that it's much easier to do:
https://github.com/parodos-dev/parodos/blob/main/workflow-service/src/main/java/com/redhat/parodos/workflow/definition/dto/WorkDefinitionResponseDTO.java#L92-L105

**Comment by nirarg** [comment]
/hold
depends on PR #289

File: notification-service-sdk/.openapi-generator/VERSION
Comment (nirarg) [reply notification-service-sdk/.openapi-generator/VERSION:1]: Moved this change to different PR
Please see PR #289

File: notification-service-sdk/pom.xml
Comment (nirarg) [reply notification-service-sdk/pom.xml:197]: Moved this change to different PR
Please see PR #289

File: .github/workflows/test.yml
Comment (nirarg) [reply .github/workflows/test.yml:39]: Adding this caused to ignore the default **surefire:test** configuration, as result of that some tests where skipped
Anyway, I changed it to exclude the regex from the pom.xml configuration
Moved this change to different PR, please see PR #290

File: Makefile
Comment (nirarg) [reply Makefile:1]: See comment above

File: notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java
Comment (nirarg) [reply notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java]: Changed
Anyway, need to change adittional places in the code (remove modelMapper usage)

File: notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java
Comment (nirarg) [reply notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java]: Fixed

File: notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java
Comment (nirarg) [reply notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java]: Changed

File: workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java
Comment (nirarg) [reply workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java]: Ok, I was looking for [org.junit.Assert.assertDoesNotThrow](https://junit.org/junit5/docs/5.8.2/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertDoesNotThrow(org.junit.jupiter.api.function.Executable))
Imported `org.junit.jupiter` library to `workflow-examples` as well

File: workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java
Comment (nirarg) [reply workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java]: Done

**Comment by nirarg** [comment]
The `Build/build` check is failing because need #290 to be merged before

Also #289 commits are included in this PR, please ignore them

Would rebase after both #290 and #289 merges

**Comment by nirarg** [comment]
I changed the integration test file name not to include the `Test` suffix
Will align the test with the changes in PR #281 after it would be merged

File: workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java
Comment (nirarg) [reply workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java]: This is integration test, each test may need to validate all operations:
1. Create notification
2. Delete notification
3. get notification
4. count notifications
Therefore, I prefer to have sub function to each operation and than each test function can call to the sub functions, and I created the logs this way to make the flow more clear.
I agree that this is shouldn't be done in unit tests, but this is not the case in this integration test

**Comment by nirarg** [comment]
/hold cancel
The PR is ready for review/merge

**Review by rgolangh — COMMENTED** [review]
/approve 
the test class could use a custom assertion instead of a function to validate but this is not mandatory

**Pull request merged**
